### ISSUE TITLE
No tag is applied for `usr.id` when UserDetails.PropagateId is true

### ISSUE THREAD

**Issue by jblazek** [DataDog/dd-trace-dotnet#3629]
**Describe the bug**

When setting `UserDetails.PropagateId=true`, there's no `usr.id` on the span. When `PropagateId=false` the `usr.id` tag is added on the current span.

I'm also not seeing `usr.id` show up anywhere when `PropagateId=true`, so I'm not sure if there's something else required for this to work.

**To Reproduce**

**Make one code path A the following code:**

```
var userDetails = new UserDetails("my-user-id-propagated");
userDetails.PropagateId = true;
tracer?.ActiveScope?.Span.SetUser(userDetails);
```

**Make a second code path B with the following code:**

```
var userDetails = new UserDetails("my-user-id-not-propagated");
tracer?.ActiveScope?.Span.SetUser(userDetails);
```

**Expected behavior**

1. Both code paths A & B apply a `usr.id` to the span
2. On child and other spans, `usr.id` gets set propagated. It doesn't seem to be anywhere I see in what gets captured.

**Screenshots**

**Runtime environment (please complete the following information):**
- Instrumentation mode: Alpine Linux extraction in Dockerfile, with datadog-dotnet-apm-2.21.0-musl.tar.gz
- Tracer version: [e.g. 2.22.1]
- OS: Alpine Linux mcr.microsoft.com/dotnet/aspnet:6.0.10-alpine3.16
- CLR: .NET 6

**Additional context**

<!--
NOTE: To contact our support team, see https://www.datadoghq.com/support/
When contacting support, please include the information requested above.
If possible, also include the tracer log files found (by default) in:
Windows: %ProgramData%\Datadog .NET Tracer\logs\
Linux: /var/log/datadog/dotnet/
--!>

**Comment by robertpi** [comment]
Hello,

In this context `PropagateId` means propagate to remote http services via the `x-datadog-tags`, it seems you're looking for a way to propagate the `usr.id` field to local child spans? We don't support propagating `usr.id` at the moment, could you tell us a little more about why you require this?

Many thanks!

**Comment by jblazek** [comment]
PropagateId is on the UserDetails type, based on the [code documentation](https://github.com/DataDog/dd-trace-dotnet/blob/57997fc09bc63a9b8a826a3400e86d2adfe72c19/tracer/src/Datadog.Trace/UserDetails.cs#L51-L59), I'd expect the `usr.id` to be propagated since the Id on `UserDetails` represents the user Id.

Regarding the use-case to require, it seems like it could be useful to propagate the user context across spans so you would be able to more easily search for a given user activity. As it works now, it seems to be a bug based on what I described above.  Setting the PropagateId=true effectively makes no `usr.id` on any span.

When you say the field means propagate to remote services via `x-datadog-tags`, what exactly gets propagated? 


Full code comment of `UserDetails.PropagateId`:
```
Gets or sets a value indicating whether the Id field should be propagated to other services called.
NOTE: setting this value to true will automatically propagate the Id field to all downstream services
including any third-party services called by the application. Therefore this value should only be set
to true if the user is confident that it is safe to propagate the value to all downstream services
called by the application.
Default value is false.
```

**Comment by pierotibou** [comment]
This has just been released as part of [2.27.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v2.27.0)
